### PR TITLE
refactoring zabbix_discovery_rule module

### PR DESCRIPTION
**Pull request by tcraxs**
##### SUMMARY
refactoring zabbix_discovery_rule module.
* use module_utils classes and functions
* add documentation for return block
* adjust return values on success

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
* zabbix_discovery_rule

##### ADDITIONAL INFORMATION
* This PR uses same changes in `helpers.py` from #119
* Have to rebase one of the PRs after one was merged, but need the changes also here to get working tests

### CODE REVIEW

**Comment by gundalow** [comment]
@tcraxs  Thank you for this PR, can you please rebase as I've just updated CI in https://github.com/ansible-collections/community.zabbix/pull/110 Thanks in advance.

File: plugins/modules/zabbix_discovery_rule.py [hunk 1]
```diff
@@ -706,26 +666,24 @@ def main():
...
 
+    drule_exists = drule.check_if_drule_exists(name)
+
     if drule_exists:
-        drule_id = zapi_wrapper.get_drule_by_drule_name(name)['druleid']
+        drule_id = drule.get_drule_by_drule_name(name)['druleid']
         if state == "absent":
             result = drule.delete_drule(drule_id)
-            module.exit_json(changed=True, result="Discovery Rule deleted: %s, ID: %s" % (name, result))
+            module.exit_json(changed=True, state=state, usergroup=name, druleid=drule_id, msg="Discovery Rule deleted: %s, ID: %s" % (name, drule_id))
```
Comment (D3DeFi) [inline plugins/modules/zabbix_discovery_rule.py:686]: prolly a typo? `usergroup=name` in multiple `exit_json` calls

File: plugins/module_utils/helpers.py [hunk 2]
```diff
@@ -27,3 +43,37 @@ def helper_cleanup_data(obj):
...
+
+def helper_to_numeric_value(strs, value):
+    """Converts string values to integers
+
+    Parameters:
+        value: string value
+
+    Returns:
+        int: converted integer
+    """
```
Comment (D3DeFi) [inline plugins/module_utils/helpers.py:56]:
```suggestion
    """
    if value is None:
        return None
```

As per `zabbix_action` module example, which will prevent few of the bug occurrences once we use this function in that module as well.

**Comment by D3DeFi** [comment]
Ready to merge after implementing those 2 suggestions above

**Comment by D3DeFi** [comment]
Than you @tcraxs !

**Pull request merged**